.github/workflows/dockerimage.yml

```diff
@@ -9,35 +9,85 @@ on:
     paths: 
       - .github/workflows/dockerimage.yml
       - build.sh
+      - deps.list
       - Dockerfile
       - protoc-wrapper
-          
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
-    - name: Build the Docker image
-      run: ./build.sh -t build
-    - name: Push the Docker image
-      if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
-      run: |
-        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
 
-        function tag_and_push {
-          docker tag build "rvolosatovs/protoc:${1}" && docker push "rvolosatovs/protoc:${1}"
-        }
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1  
+      with:
+        install: true
+
+    - name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v1
+      with:
+        image: tonistiigi/binfmt:latest
+
+    - name: Docker meta
+      id: docker_meta
+      uses: docker/metadata-action@v3
+      with:
+        images: strophy/protoc
```

Comment:
This should be updated, as you already mentioned in the PR description, just adding a comment here to not forget to fix before merge

Reply:
This still needs to be updated